### PR TITLE
Added versions api & tests

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Added versions api

1) Versions list - all versions - `/api/v1/versions/`
2) Version Detail - Get specific version by Id - `/api/v1/version/<id>/`
3) all Versions available for a specific profile - `/api/v1/profile/<id>/versions/`

## Related Issue
Not related to any specific issue.
I think I was overthinking and made an API endpoint so we can actually see and create a version through API user if required.
This PR can be closed if not required.
Now I can see that we pass versions in profile api

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: tests/datasets/test_views.py [hunk 1]
```diff
@@ -282,3 +282,45 @@ def test_updating_dataset_with_indicator_update_and_overwrite(self):
...
+    def test_version_by_profile_view(self):
+        response = self.get(
+            'versions-by-profile', profile_id=self.profile.id,
+            extra={'format': 'json'}
+        )
+
+        assert response.status_code == 200
+        data = response.data
+        assert data["count"] == 1
+        assert data["results"][0]["id"] == self.version1.id
```
Comment (michaelglenister) [inline tests/datasets/test_views.py:326]: I don't think it can hurt to check the version name is also set

**Review by michaelglenister — COMMENTED** [review]
If you can also fix the merge conflict, then it should be ready to merge, thanks

**Comment by goyal1092** [comment]
updated PR @michaelglenister 
Thanks for review :)

**Comment by jbothma** [comment]
No need for this endpoint as yet.

**Pull request closed without merging**